### PR TITLE
Add the idrac-redfish-virtual-media boot interface

### PR DESCRIPTION
**Pull request by juliakreger**
Note: This boot interface requires an addtiaional sushy-oem-dell
package which is not yet available as an RPM.

### CODE REVIEW

**Comment by juliakreger** [comment]
So what we're missing here is the sushy-oem-dell package. The change to RDO mirrored, although seems not to be packaged yet. I guess we could git clone and install locally initially?

@derekhiggins Seems like an easy compromise to just manually install it from git for now?

**Comment by elfosardo** [comment]
My advice is to install from pypi if the version there is what we need.
If we need a more recent commit, then use pip to install from the git repo, no need for clone.

**Comment by dtantsur** [comment]
The first package will be there once somebody from RDO approves https://review.rdoproject.org/r/#/c/24057/. You can even use the packages from the CI job: https://logs.rdoproject.org/57/24057/9/check/DLRN-rpmbuild/b26763b/buildset/centos/current/

**Comment by derekhiggins** [comment]
> So what we're missing here is the sushy-oem-dell package. The change to RDO mirrored, although seems not to be packaged yet. I guess we could git clone and install locally initially?
> 
> @derekhiggins Seems like an easy compromise to just manually install it from git for now?

Looks like the package is now available in the RDO train repository, so you should be able to just install it
https://trunk.rdoproject.org/train/centos7/current/

**Comment by derekhiggins** [comment]
Looks like we need to either move to rdo/master or back port https://review.opendev.org/#/c/672498/ , trying this out I got an error about no entry point for idrac-redfish-virtual-media

**Comment by juliakreger** [comment]
@derekhiggins I believe we've already backported, it is just not going to make it out into the packaging pipeline for weeks from what I understood from the packaging/release team.

**Comment by derekhiggins** [comment]
> @derekhiggins I believe we've already backported, it is just not going to make it out into the packaging pipeline for weeks from what I understood from the packaging/release team.

Do you mean backported downstream? I don't see it in upstream ironic stable/train which is what this image is using.

**Comment by dtantsur** [comment]
The package is ready in RDO. However, we cannot really backport that feature upstream to Train. Any suggestions? Should we switch to patching code in-place (SIGH)?

**Comment by elfosardo** [comment]
we will have to start using master again then
I would avoid patching code in-place

**Comment by dtantsur** [comment]
I don't think switching to master is currently on the radar, especially because of the troubles it would cause downstream..

**Comment by elfosardo** [comment]
@dtantsur  @juliakreger  @derekhiggins  #125 should add the dependency

**Comment by dtantsur** [comment]
We'll also need https://github.com/metal3-io/ironic-image/pull/100 to be able to switch to master.

**Comment by elfosardo** [comment]
We should be ready to move forward with this now

**Comment by juliakreger** [comment]
It can't be tried until it merges, well aside from the work @derekhiggins did.

File: ironic.conf [hunk 1]
```diff
@@ -5,7 +5,7 @@ default_boot_interface = ipxe
 default_deploy_interface = direct
 default_inspect_interface = inspector
 default_network_interface = noop
-enabled_boot_interfaces = pxe,ipxe,fake,redfish-virtual-media
+enabled_boot_interfaces = pxe,ipxe,fake,redfish-virtual-media,idrac-redfish-virtual-media
 enabled_deploy_interfaces = direct,fake
 enabled_hardware_types = ipmi,idrac,irmc,fake-hardware,redfish
```
Comment (dtantsur) [inline ironic.conf]: This needs `idrac-redfish`, no?

Comment (juliakreger) [reply ironic.conf]: was already present, looks like I need to rebase.

**Pull request merged**